### DISCUSSION TITLE
Rendering videos at scale using lambda

### DISCUSSION THREAD

**Discussion by reactone** [remotion-dev/remotion#1424] [General]
I am wondering what is the most scalable approach regarding rendering videos. Currently I am rendering the video locally by passing json file as props using a node endpoint (passing json file) which triggers the render with json as input on my server or locally in the terminal. My concern is that if I have 100 users rendering the video at the same time obviously it wont work. I was looking into lambda render on AWS but I read that the maximum payload is only 128 KB which is not enough as my json file contains all the components that need to be rendered and that could be several megabytes in size. What is the best solution to render videos at scale? Im I missing something? Is passing the props the right solution or should it be done differently?

**Comment by JonnyBurger** [comment]
~Hi @reactone!~

~I think Lambda is the best solution we have and regarding the 128KB limit - it concerns how big the serialized `inputProps` may be at most.~

~Your React components can be much bigger and even if just your inputProps are bigger than that, you can save a JSON for example in your S3 bucket and pass a link as the input props, and then download the payload inside your component. We have to use this workaround because that is an AWS limit, but if we use this workaround, then we have no limits :)~

**Comment by JonnyBurger** [comment]
Forget what I said - I realized we can handle this in Remotion itself - from next version on, there will be no more inputProps payload limit! 🎉 https://github.com/remotion-dev/remotion/pull/1430

**Comment by reactone** [comment]
Thats great, were you able to find a workaround? Are there some additional steps required in order for this to work?

**Reply by JonnyBurger** [reply]
@reactone All you need to do is to upgrade once I release the next version.
If your input props exceed 256KB, we'll upload them to S3 and then download them from the Lambda function itself again.
